packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -647,18 +647,21 @@ export class ConnectionManager implements IConnectionManager {
 
 				const waitStartTime = performance.now();
 				const retryDelayFromError = getRetryDelayFromError(origError);
-				if (retryDelayFromError !== undefined) {
-					// If the error told us to wait, then we wait.
-					this.props.reconnectionDelayHandler(retryDelayFromError, origError);
+				// If the error told us to wait, then we wait for that time. If the error didn't tell us to wait,
+				// let's still wait a little bit before retrying. We skip this delay if we're confident we're offline,
+				// because we probably just need to wait to come back online. But we never have strong signal of being
+				// offline, so we at least wait for sometime.
+				if (retryDelayFromError !== undefined || globalThis.navigator?.onLine !== false) {
+					delayMs = Math.max(retryDelayFromError ?? 0, delayMs);
+					// Raise event in case the delay was there.
+					this.props.reconnectionDelayHandler(delayMs, origError);
 					await new Promise<void>((resolve) => {
-						setTimeout(resolve, retryDelayFromError);
+						setTimeout(resolve, delayMs);
 					});
-				} else if (globalThis.navigator?.onLine !== false) {
-					// If the error didn't tell us to wait, let's still wait a little bit before retrying.
-					// We skip this delay if we're confident we're offline, because we probably just need to wait to come back online.
+					delayMs = Math.min(delayMs * 2, calculateMaxWaitTime(origError));
+				} else {
 					await new Promise<void>((resolve) => {
-						setTimeout(resolve, delayMs);
-						delayMs = Math.min(delayMs * 2, calculateMaxWaitTime(origError));
```

Comment:
Suggestion - reduce duplication and move it outside of IF/ELSE - both branches do the same thing